### ISSUE TITLE
SASS Styling Standard For Loconomics-Pages

### ISSUE THREAD

**Issue by tkwok** [loconomics/loconomics-pages#20]
@IagoSRL Is it possible that we follow a SASS convention that everyone can be agreed on, for current and future growth? 

Here is a standard that I think is functional and can be followed easily, which includes folder structure, file naming, and best-practices.

https://sass-guidelin.es/

Feel free to share your thoughts about this. Thanks.

**Comment by IagoSRL** [comment]
Hi @tkwok, agree that a coding style is good. On my work here I followed the [convention in use at the App](https://github.com/loconomics/loconomics/blob/master/docs/Contributor%20Guidelines.md) at the main repo (loconomics/loconomics) but is simple, few details as 4 spaces and SUIT for naming classes.

From a quick read of `sass-guidelin`, it looks I already followed several details of the convention without knowledge! (like folders, usage of underscore in filenames, scope of variables, some line org like priorities -mixims first, white lines); but I see I used a more JavaScript convention for variables names that doesn't match this and actually is different than Bootstrap one, so I will change that (from `$outerColor` to `$outer-color`).

### Class naming convention
About the naming convention for CSS classes, as set in the Contributor Guidelines, we used SUIT, that fits well the idea of components (complex, usually nested classes, matching a JavaScript class name when has scripting).
The `sass-guidelin` explicitly lefts [out of scope](https://sass-guidelin.es/#naming-conventions) the decision of a *class naming convention*, so it does not collides with what we have already.

### What about BEM and Boostrap?
While both SUIT and the popular BEM convention are good, the usage of SUIT removes confusion in a project that uses Bootstrap classes too.
Bootstrap does not follow any of that conventions, their own one is pretty simple, doesn't define hierarchy that sometimes leads to problems (being usage of `row` classes out of place the one I saw more) and has not clear separation between classes (like modifiers). Because of that, I think the Bootstrap style doesn't scale well, so we are forced to choose another one for our code and to have a mix of two styles on class usage.

At SUIT, names start with an uppercase, so is easy to distinguish a class in that convention from a Bootstrap class (all lowercase), and you know better where its definition can be found and the hierarchy followed at children/ancestors elements. BEM uses all lowercase too, so at first shoot at a class name you don't know which style follows (if comes from Bootsrap or our code).
And even prevent collides if found that Boostrap adds a new class name that matches one we created already (think on the new `card` at BS4 and our own `Card` component -it's a situation that needs a name change of course, but almost when initially introduced the new version the styles will not break, and at some point we will migrate the App to BS4).
So basically, SUIT give as a more clear code in a situation we are forced to have a mix of styles.

-----------------

Please, let me your thoughts!
